todo-app/src/hooks/useStorageValue.js

```diff
@@ -0,0 +1,12 @@
+import { useState, useEffect } from 'react';
+import { getStorageValue } from '../helpers/getStorageValue';
+
+export const useLocalStorage = (key, defaultValue) => {
+  const [value, setValue] = useState(() => getStorageValue({ key, defaultValue }));
```

Comment:
Not sure about this pattern, what is this expected to do?
I would imagine you need to call the function here like this
```suggestion
  const [value, setValue] = useState(getStorageValue({ key, defaultValue }));
```
for this to work?
At least I can't seem to find a reference to the ability to put a function for the initialState in the react docs.

Reply:
Ohhhh TIL 🤩 

I knew of it for the `setState` part, not for the initial state, very cool!

One more small thing to implement in our own version 🤩